File: piAmine/src/entities/Activite.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;


import java.sql.Date;
import java.time.LocalDate;
/**
 *
 * @author thebejaoui
 */
public class Activite {
     private int ida;
    private int id_prop;
        private String libele;
   private float prix;
   private String emplacement;
   private String description;
   private Date  date;
   private String  type;
   private String avis;

    public Activite( int id_prop, String libele, float prix, String emplacement, String description, Date date, String type, String avis) {
        
        this.id_prop = id_prop;
        this.libele = libele;
        
        this.prix = prix;
        this.emplacement = emplacement;
        this.description = description;
        this.date = date;
        this.type = type;
        this.avis = avis;
    }

    public Activite(int ida) {
        this.ida = ida;
    }

    public Activite(int ida, int id_prop, String libele,float prix, String emplacement, String description, Date date, String type, String avis) {
        this.ida = ida;
        this.id_prop = id_prop;
        this.libele = libele;
        this.prix = prix;
        this.emplacement = emplacement;
        this.description = description;
        this.date = date;
        this.type = type;
        this.avis = avis;
    }

  

   
   

    public int getIda() {
        return ida;
    }

    public void setIda(int ida) {
        this.ida = ida;
    }

    public int getIdu() {
        return id_prop;
    }

    public void setIdu(int id_prop) {
        this.id_prop = id_prop;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public String getEmplacement() {
        return emplacement;
    }

    public void setEmplacement(String emplacement) {
        this.emplacement = emplacement;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getAvis() {
        return avis;
    }

    public void setAvis(String avis) {
        this.avis = avis;
    }

    public String getLibele() {
        return libele;
    }

    public void setLibele(String libele) {
        this.libele = libele;
    }

    
    @Override
    public String toString() {
        return "Activite{" + "ida=" + ida + ", idu=" + id_prop + ", prix=" + prix + ", emplacement=" + emplacement + ", description=" + description + ", date=" + date + ", type=" + type + ", avis=" + avis + '}';
    }

  

    
   
   
   
    
    
}

File: piweb/src/Entity/Activite.php
<?php

namespace App\Entity;

use App\Repository\ActiviteRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=ActiviteRepository::class)
 */
class Activite
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @Assert\NotBlank(message=" Libele est obligatoire")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer un Libele au mini de 4 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $libele;

    /**
     * @Assert\NotBlank(message=" Prix est obligatoire")
     * @ORM\Column(type="float")
     * @Assert\Type(type="float")
     */
    private $prix;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $emplacement;


    /**
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=100" )
     * @ORM\Column(type="string", length=1000)
     */
    private $description;

    /**
     * @ORM\Column(type="date")
     */
    private $date;

    /**
     *
     * @Assert\NotBlank(message=" Type est obligatoire")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer un Libele au mini de 5 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $type;

    /**
     * @Assert\NotBlank(message=" Avis est obligatoire")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer un Libele au mini de 5 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $avis;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $image;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLibele(): ?string
    {
        return $this->libele;
    }

    public function setLibele(string $libele): self
    {
        $this->libele = $libele;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getEmplacement(): ?string
    {
        return $this->emplacement;
    }

    public function setEmplacement(string $emplacement): self
    {
        $this->emplacement = $emplacement;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getAvis(): ?string
    {
        return $this->avis;
    }

    public function setAvis(string $avis): self
    {
        $this->avis = $avis;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }
}
